src/Controller/EnMarche/AdherentMessage/CommitteeMessageController.php

```diff
@@ -2,17 +2,263 @@
+     */
+    public function messageListAction(
+        Request $request,
+        UserInterface $adherent,
+        AdherentMessageRepository $repository,
+        Committee $committee = null
+    ): Response {
+        $this->disableInProduction();
+
+        $status = $request->query->get('status');
+
+        if ($status && !AdherentMessageStatusEnum::isValid($status)) {
+            throw new BadRequestHttpException('Invalid status');
+        }
+
+        return $this->renderTemplate('message/list.html.twig', [
+            'messages' => $repository->findAllCommitteeMessage($adherent, $committee, $status),
+            'committee' => $committee,
+            'route_params' => ['committee_slug' => $committee->getSlug()],
+        ]);
+    }
+
+    /**
+     * @Route("/creer", name="create", methods={"GET", "POST"})
+     *
+     * @param Adherent|UserInterface $adherent
+     */
+    public function createMessageAction(
+        Request $request,
+        UserInterface $adherent,
+        ObjectManager $manager,
+        MessageBusInterface $bus,
+        Committee $committee = null
+    ): Response {
+        $this->disableInProduction();
+
+        $form = $this
+            ->createForm(AdherentMessageType::class)
+            ->handleRequest($request)
+        ;
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $message = AdherentMessageFactory::create($adherent, $form->getData(), $this->getMessageType());
+            $message->setFilter(new CommitteeFilter($committee));
+
+            $manager->persist($message);
+
+            $manager->flush();
+
+            $this->addFlash('info', 'adherent_message.created_successfully');
+
+            if ($form->get('next')->isClicked()) {
+                return $this->redirectToMessageRoute('filter', [
+                    'uuid' => $message->getUuid()->toString(),
+                    'committee_slug' => $committee->getSlug(),
+                ]);
+            }
+
+            return $this->redirectToMessageRoute('update', [
+                'uuid' => $message->getUuid(),
+                'committee_slug' => $committee->getSlug(),
+            ]);
+        }
+
+        return $this->renderTemplate('message/create.html.twig', [
+            'form' => $form->createView(),
+            'committee' => $committee,
+            'route_params' => ['committee_slug' => $committee->getSlug()],
+        ]);
+    }
+
+    /**
+     * @Route("/{uuid}/modifier", requirements={"uuid": "%pattern_uuid%"}, name="update", methods={"GET", "POST"})
+     *
+     * @Security("is_granted('IS_AUTHOR_OF', message)")
+     */
+    public function updateMessageAction(
+        Request $request,
+        AbstractAdherentMessage $message,
+        ObjectManager $manager,
+        Committee $committee = null
+    ): Response {
+        $this->disableInProduction();
+
+        if ($message->isSent()) {
+            throw new BadRequestHttpException('This message has been already sent.');
```

Comment:
Hmmm not sure but `This message has already been sent` sounds better to me :thinking: ?

Reply:
done